topic-operator/src/main/java/io/strimzi/operator/topic/BatchingTopicController.java

```diff
@@ -1069,106 +1075,168 @@ private Collection<AlterConfigOp> buildAlterConfigOps(Reconciliation reconciliat
         HashSet<String> keysToRemove = configs.entries().stream()
                 .filter(configEntry -> configEntry.source() == ConfigEntry.ConfigSource.DYNAMIC_TOPIC_CONFIG)
                 .map(ConfigEntry::name).collect(Collectors.toCollection(HashSet::new));
-        if (hasConfig(kt)) {
-            keysToRemove.removeAll(kt.getSpec().getConfig().keySet());
+        if (hasConfig(reconcilableTopic.kt())) {
+            keysToRemove.removeAll(reconcilableTopic.kt().getSpec().getConfig().keySet());
         }
         for (var key : keysToRemove) {
             alterConfigOps.add(new AlterConfigOp(
                     new ConfigEntry(key, null),
                     AlterConfigOp.OpType.DELETE));
         }
 
-        skipNonAlterableConfigs(alterConfigOps);
-
         if (alterConfigOps.isEmpty()) {
-            LOGGER.debugCr(reconciliation, "No config change");
+            LOGGER.debugCr(reconcilableTopic.reconciliation(), "No config change");
         } else {
-            LOGGER.debugCr(reconciliation, "Config changes {}", alterConfigOps);
+            LOGGER.debugCr(reconcilableTopic.reconciliation(), "Config changes {}", alterConfigOps);
         }
         return alterConfigOps;
     }
 
     /**
-     * <p>The Topic Operator {@code alterableTopicConfig} can be used to specify a comma separated list of Kafka
-     * topic configurations that can be altered by users through {@code .spec.config}. Keep in mind that if changes
-     * are applied directly in Kafka, the operator will try to revert them producing a warning.</p>
-     *
-     * <p>This is useful in standalone mode when you have a Kafka service that restricts alter operations
-     * to a subset of all the Kafka topic configurations.</p>
+     * Check if there is any {@code .spec.config} property that should be ignored.
+     * If found, any pending alter config operation for that property is removed and a warning is raised.
      *
+     * @param reconcilableTopics Topics tha may need update.
+     * @param alterConfigPairs Detected alter config operations.
+     */
+    private void checkSpecConfigs(List<ReconcilableTopic> reconcilableTopics,
```

Comment:
If there any need for the `alterConfigPairs` to be mutated by this method (actually the `check*Config()` methods)? It would be more functional to just return a filtered list, and not modifying the parameter.

Reply:
Thanks for raising. I was more focused on correctness, but now I fixed that. I also moved the config changes debug logs at the end of the filter logic. Let me know if it looks good now.